app/models/session.rb

```diff
@@ -0,0 +1,19 @@
+class Session < ActiveRecord::Base
+
+  def to_h
+    data.dup
+  end
+
+  # rubocop:disable Style/MethodMissing
+  def method_missing(m, *args)
```

Comment:
I'm not a big fan of the `method_missing` unless it's being used in a framework or lib. I understand this implementation will give us setters/getters dynamically and have something like 
```
obj = Session.new
obj.name='Jane'
obj.post_code='E1'
obj.crazy_field='something crazy'
```
but you lose control of the Session's API, making hard to track misspelled calls like
```
obj.to_hh #data look up instead of warning me about a wrong method
=> nil
```
which can be worse if the class gets more methods.

Shouldn't something like the following be still simple enough and less error prone?
```
obj = Session.new
obj.data = {name: 'Jane', post_code: 'E1'} #set anything you like here
obj.add_data({crazy_field: 'something crazy'})
obj.data 
=> {:name => 'Jane', :post_code => 'E1', :crazy_field => 'something crazy'}
obj.data[:name]
=> "Jane"
```

Reply:
If we really like the `obj.name = 'Jane'` solution, then maybe it would worth consider using `OpenStruct`?

Reply:
Also, the class name is confusing. Maybe rename it to something like TempStorage? - I'm sure you can come up with a better name than me :)

Reply:
We can't use open struct because it needs to be an AR object that persists to the DB. So I have just used a standard active record class that uses method missing so that you can get and set anything and persist to the database.  If you call the wrong method, you will get the wrong value - just like the underlying hash would.

We could make this so it us purely a 'Temporary Refund' and not general purpose and put the getters ad setters in - then rename the model and table name to suit.
But, would we not then be better off just having a table with columns again, but no DB level validation for null etc.. Even then, you  would look at the class and it would be empty just like normal AR classes without any validation etc...

Reply:
I totally agree about the naming - when you first mentioned it, I thought 'Hmm, thats not a very good name really'.

Ill rename when we know if we want to change it to be more specific or not

Reply:
With the above method, we can add and remove columns from the refund schema etc.. without having to change this at all

Reply:
Sorry, my example wasn't very clear. I meant to convert the data variable into an OpenStruct obj which would allow us to do `obj.data.name = 'Jane'` and `obj.data.name #=> Jane`.  If AR complains when saving it, then saving the `data.to_h` should do. Nevertheless, I would still prefer to use the simple and self explanatory `obj.data = {name: 'Jane'}`. 

Also, this method_missing will not only spend time and resources for the method lookups, but will confuse devs when calling wrong methods by returning nil values instead of error messages.

Now, all the above are just my thoughts around the method_missing and how to avoid issues, but if you think the possibilities of getting in trouble with the current implementation are low, then I am happy to keep it as is. :)

Reply:
I know Ive replied to this on slack - but just for the record.  The form objects are written to use AR objects where the attributes are methods on the object and the save method is on the same object, also the valid? method etc...  I think it would get very hairy if we went down that route